zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/znode/groupacl/X509ZNodeGroupAclProvider.java

```diff
@@ -238,8 +238,13 @@ private void assignAuthInfo(ServerCnxn cnxn, String clientId, Set<String> domain
         cnxn.close(ServerCnxn.DisconnectReason.SSL_AUTH_FAILURE);
       }
     } else {
-      // Assign Auth Id according to domains
-      domains.stream().forEach(d -> newAuthIds.add(new Id(getScheme(), d)));
+      domains.forEach(d -> {
+        if (superUserDomainNames.contains(d)) {
+          newAuthIds.add(new Id(X509AuthenticationUtil.SUPERUSER_AUTH_SCHEME, d));
```

Comment:
Let's clarify what kind of special treatment we are doing for non-superusers / cross-domain components and why?